src/main/java/org/folio/circulation/domain/LocationRepository.java

```diff
@@ -18,33 +16,72 @@
 import org.folio.circulation.support.Result;
 import org.folio.circulation.support.SingleRecordFetcher;
 
-import io.vertx.core.json.JsonObject;
-
 public class LocationRepository {
+
   private CollectionResourceClient locationsStorageClient;
+  private CollectionResourceClient institutionsStorageClient;
+  private CollectionResourceClient campusesStorageClient;
+  private CollectionResourceClient librariesStorageClient;
 
-  public LocationRepository(Clients clients) {
-    locationsStorageClient = clients.locationsStorage();
+  public LocationRepository(CollectionResourceClient locationsStorageClient,
+                            CollectionResourceClient institutionsStorageClient,
+                            CollectionResourceClient campusesStorageClient,
+                            CollectionResourceClient librariesStorageClient) {
+    this.locationsStorageClient = locationsStorageClient;
+    this.institutionsStorageClient = institutionsStorageClient;
+    this.campusesStorageClient = campusesStorageClient;
+    this.librariesStorageClient = librariesStorageClient;
   }
 
-  public CompletableFuture<Result<JsonObject>> getLocation(Item item) {
+  public static LocationRepository using(Clients clients) {
+    return new LocationRepository(
+      clients.locationsStorage(),
+      clients.institutionsStorage(),
+      clients.campusesStorage(),
+      clients.librariesStorage()
+    );
+  }
+
+  public CompletableFuture<Result<Location>> getLocation(Item item) {
```

Comment:
Please can all endpoints that use this method have the relevant extra permissions added for fetching the location hierarchy